### ISSUE TITLE
`nil` values for attributes not syncing to iCloud (CoreData)

### ISSUE THREAD

**Issue by Kautenja** [mentrena/SyncKit#10]
I'm seeing an issue with syncing relationships when values are removed / nullified.

consider the example case

```swift
class Parent {
    @NSManaged var children: NSSet
}

class Child {
    @NSManaged var parent: Parent?
}
```

if for some arbitrary instance of child parent is set

```swift
let arb = Child(context: ...)
arb.parent = someInstanceOfParent
```

and saved, this will sync fine. If it's changed to a different value and saved again it syncs fine etc. The problem is when it is set to nil on one device and synchronized to another, this change is not synchronized to the local core data store and the last set value persists.

The problem exists from the other direction as well where if an object is removed from an NSSet and synchronized, the changes are not reflected in the iCloud store.

I've verified this with both one-to-one and one-to-many relationships. nil values just don't stick to the cloud store. Note that in these cases the objects are not being deleted, the relationship is just being nullified.

TL;DR
* when a relationship is set to nil, or an object is removed from a relationship set and saved, the changes _do not_ synchronize correctly with iCloud

**Comment by Kautenja** [comment]
Edit: hadn't noticed, but this is not just the case for relationships, but also plain values. for instance:

```swift
class Thing {
    var someDate: NSDate?
}
```

if i set some arbitrary value of some arbitrary object to a _non nil value_ then save it to the iCloud store and then repeat the process setting the value to a _nil_ value, the change is not reflected in the iCloud store

**Comment by mentrena** [comment]
Hi @Kautenja ,

Thanks for reporting this. 
I ran a quick test and I can see the change being uploaded to iCloud. However, when the second device downloaded the changes, these didn't include the nil field. I'll need to investigate further...

**Comment by mentrena** [comment]
Hopefully it should be fixed on `0.4.6`

**Comment by Kautenja** [comment]
Can confirm the fix in all instances where I had seen it! thanks for all your work on this framework, it's quite handy :)
